src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+                }
+            }
+
+            Console.WriteLine($"Found {fixes.Count} potential code fixes");
+
+            Console.WriteLine("Calculating changes");
+
+            stopwatch.Restart();
+
+            object lockObject = new object();
+
+            Parallel.ForEach(fixes, fix =>
+            {
+                try
+                {
+                    fix.GetOperationsAsync(cancellationToken).GetAwaiter().GetResult();
+                }
+                catch (Exception ex)
+                {
+                    // Report thrown exceptions
+                    lock (lockObject)
+                    {
+                        Utilities.WriteLine($"The fix '{fix.Title}' threw an exception:", ConsoleColor.Yellow);
+                        Utilities.WriteLine(ex.ToString(), ConsoleColor.Red);
+                    }
+                }
+            });
+
+            Console.WriteLine($"Calculating changes completed in {stopwatch.ElapsedMilliseconds}ms");
+        }
+
+        private static async Task<IEnumerable<CodeAction>> GetFixesAsync(Solution solution, CodeFixProvider codeFixProvider, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            List<CodeAction> codeActions = new List<CodeAction>();
+
+            await codeFixProvider.RegisterCodeFixesAsync(new CodeFixContext(solution.GetDocument(diagnostic.Location.SourceTree), diagnostic, (a, d) => codeActions.Add(a), cancellationToken)).ConfigureAwait(false);
+
+            return codeActions;
+        }
+
+        private static Task<Statistic> GetAnalyzerStatisticsAsync(IEnumerable<Project> projects, CancellationToken cancellationToken)
+        {
+            ConcurrentBag<Statistic> sums = new ConcurrentBag<Statistic>();
+
+            Parallel.ForEach(projects.SelectMany(i => i.Documents), document =>
+            {
+                var documentStatistics = GetAnalyzerStatisticsAsync(document, cancellationToken).ConfigureAwait(false).GetAwaiter().GetResult();
+                sums.Add(documentStatistics);
+            });
+
+            Statistic sum = sums.Aggregate(new Statistic(0, 0, 0), (currentResult, value) => currentResult + value);
+            return Task.FromResult(sum);
+        }
+
+        private static async Task<Statistic> GetAnalyzerStatisticsAsync(Document document, CancellationToken cancellationToken)
+        {
+            SyntaxTree tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+
+            SyntaxNode root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var tokensAndNodes = root.DescendantNodesAndTokensAndSelf(descendIntoTrivia: true);
+
+            int numberOfNodes = tokensAndNodes.Count(x => x.IsNode);
+            int numberOfTokens = tokensAndNodes.Count(x => x.IsToken);
+            int numberOfTrivia = root.DescendantTrivia(descendIntoTrivia: true).Count();
+
+            return new Statistic(numberOfNodes, numberOfTokens, numberOfTrivia);
```

Comment:
this code path might be good place to show results of GetDiagnostics I mentioned above. not this particular one, since GetDiagnostics are for Project, not Document.

and here I meant Compilation.GetDiagnostics. not CompilationWithAnalyzer.GetAnalysisResultAsync

Reply:
➡️ Let's file/examine this as a follow-up issue, after we can get our hands on the first build and play with it some.

Reply:
[https://github.com/dotnet/roslyn/issues/23108](https://github.com/dotnet/roslyn/issues/23108)

---
In reply to: [149861217](https://github.com/dotnet/roslyn/pull/23087#discussion_r149861217) [](ancestors = 149861217)